packages/devtools_server/lib/src/server.dart

```diff
@@ -92,14 +97,82 @@ void serveDevTools({
 
   final server = await shelf.serve(handler, '127.0.0.1', port);
 
+  final devtoolsUrl = 'http://${server.address.host}:${server.port}';
+
   printOutput(
-    'Serving DevTools at http://${server.address.host}:${server.port}',
+    'Serving DevTools at $devtoolsUrl',
     {
       'method': 'server.started',
       'params': {'host': server.address.host, 'port': server.port, 'pid': pid}
     },
     machineMode: machineMode,
   );
+
+  _listenForUris(devtoolsUrl);
+}
+
+void _listenForUris(String devtoolsUrl) {
```

Comment:
I would just move this back into the main method. I'm not seeing any extra value in separating it from main. The helper that would be more useful for testing is a helper that takes a single VMService WS URI